flink-runtime/src/main/java/org/apache/flink/runtime/io/network/api/serialization/SpillingAdaptiveSpanningRecordDeserializer.java

```diff
@@ -630,8 +630,7 @@ private FileChannel createSpillingChannel() throws IOException {
 			}
 
 			// try to find a unique file name for the spilling channel
-			int maxAttempts = 10;
-			for (int attempt = 0; attempt < maxAttempts; attempt++) {
+			for (int attempt = 0; attempt < tempDirs.length; attempt++) {
```

Comment:
I am not clear what is the motivation for changing this. 

The following file name generation is not only based on `tempDirs.length` which is only for the parent dir, and the specific sub name is from `randomString(rnd)`, so I think it does not very make sense to limit the retry times directly based on `tempDirs.length`.

Reply:
The mount disk should exist and this logic is for creating unique random file for spilling, so we add the retry times to avoid random file which is already created before to cause failure here. From this point, it seems not to get any benefits by changing the constant retry times.